sync/awsdatabricks.py

```diff
@@ -227,64 +232,67 @@ def _get_cluster_instances(cluster: dict) -> Response[dict]:
     #  are associated with this cluster
-    if not cluster_instances:
+    if not cluster_info:
         ec2 = boto.client("ec2", region_name=aws_region_name)
         try:
-            cluster_instances = ec2.describe_instances(
-                Filters=[
-                    {"Name": "tag:Vendor", "Values": ["Databricks"]},
-                    {"Name": "tag:ClusterId", "Values": [cluster_id]},
-                    # {'Name': 'tag:JobId', 'Values': []}
-                ]
-            )
+            cluster_info = {
+                "Reservations": _get_ec2_instances(cluster_id, ec2),
+                "Volumes": _get_ebs_volumes(cluster_id, ec2),
+            }
+
         except Exception as exc:
             logger.warning(exc)
 
-    if not cluster_instances or not cluster_instances["Reservations"]:
-        no_instances_message = (
-            f"Unable to find any active or recently terminated instances for cluster `{cluster_id}` in `{aws_region_name}`. "
+    def missing_message(input: str) -> str:
+        return (
+            f"Unable to find any active or recently terminated {input} for cluster `{cluster_id}` in `{aws_region_name}`. "
             + "Please refer to the following documentation for options on how to address this - "
             + "https://docs.synccomputing.com/sync-gradient/integrating-with-gradient/databricks-workflows"
         )
-        return Response(error=DatabricksError(message=no_instances_message))
 
-    return Response(result=cluster_instances)
+    if not cluster_info or not cluster_info["Reservations"]:
+        reservations_response = Response(
+            error=DatabricksError(message=missing_message("instances"))
+        )
+    else:
+        reservations_response = Response(result={"Reservations": cluster_info["Reservations"]})
+
+    if not cluster_info or not cluster_info["Volumes"]:
+        volumes_response = Response(error=DatabricksError(message=missing_message("ebs volumes")))
+    else:
+        volumes_response = Response(result={"Volumes": cluster_info["Volumes"]})
 
+    return reservations_response, volumes_response
 
-def _get_cluster_instances_from_s3(bucket: str, file_key: str, cluster_id):
+
+def _get_aws_cluster_info_from_s3(bucket: str, file_key: str, cluster_id):
     s3 = boto.client("s3")
     try:
         return s3.get_object(Bucket=bucket, Key=file_key)["Body"].read()
     except ClientError as ex:
         if ex.response["Error"]["Code"] == "NoSuchKey":
             logger.warning(
-                f"Could not find sync_data/cluster_instances.json for cluster: {cluster_id}"
+                f"Could not find sync_data/aws_cluster_info.json for cluster: {cluster_id}"
             )
         else:
             logger.error(
-                f"Unexpected error encountered while attempting to fetch sync_data/cluster_instances.json: {ex}"
+                f"Unexpected error encountered while attempting to fetch sync_data/aws_cluster_info.json: {ex}"
             )
```

Comment:
Potential improvement:
```
except ClientError as err:
    logger.warning(f"Failed to retrieve cluster info from S3 with key, '{file_key}': {err}")
```
The reason for the exception is not important to the consumer which anticipates a `None` response so it should be a warning in either case. If we need to better understand the reason it'll be in the log statement, e.g. "Failed to retrieve cluster info from S3 with key, 'sync_data/1234210239561239/aws_cluster_info.json': An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist."

Reply:
Done! Happy to make changes that cut out lines :D